http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesSprayJsonCodec.scala

```diff
@@ -145,6 +145,14 @@ object LeoRoutesSprayJsonCodec extends DefaultJsonProtocol {
     }
   }
 
+  implicit val runtimeStatusFormat: RootJsonWriter[RuntimeStatus] = (status: RuntimeStatus) => {
```

Comment:
Could we put this logic in circe `JsonCodec`? Have been trying to not add new spray encoders -- this is just used for the legacy routes anyway, the new `/runtimes` routes use all circe encoding/decoding.

Reply:
this is just to support the old route...new route uses circle (https://github.com/DataBiosphere/leonardo/pull/1395/files#diff-9bef6acafa788ef59ee57f5d1618439aR329)...I didn't put it in `JsonCodec` in core is because this encoding is contract specific...most likely anywhere else we don't want to do this encoding (for instance, in Leo pubsub, it's encoded differently)

Reply:
Yeah I see now.. thanks

Reply:
I'd prefer not to make frontend do publishing to google synchronous to API calls everywhere. If we're worried about the delay in UI, I think a better way is to introduce more `pre` statuses persisted synchronous to API call

Reply:
does this belong in the other thread? :)

I just think the transactional nature of these APIs is a contract that we shouldn't break. `pre` statuses could be a solution. Or maybe a flag `isPublished` in the DB which is initially `false` but flipped to `true` once the google msg has been published.

Reply:
oops yea...got threads mixed up.

I kinda prefer `pre`, cuz we'll still need `PreCreating` even if we introduce `isPublished`...So I feel introducing `Pre` for all transition requests seems more consistent.